File: app/src/main/java/com/devdream/cookall/core/api/responses/BaseResponse.java
package com.devdream.cookall.core.api.responses;

import com.google.gson.annotations.SerializedName;

public abstract class BaseResponse<T> {

    @SerializedName("code")
    private int code;

    @SerializedName("status")
    private String status;

    @SerializedName("data")
    private T data;

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

}

File: app/src/main/java/com/devdream/cookall/core/api/responses/BaseResponse.kt
package com.devdream.cookall.core.api.responses

import com.google.gson.annotations.SerializedName

abstract class BaseResponse<T> {

    @SerializedName("code")
    var code: Int = 0

    @SerializedName("status")
    var status: String? = null

    @SerializedName("data")
    var data: T? = null

}
